### PR TITLE
programs.zsh: generate a more compact config

### PR DESCRIPTION
**Pull request by teto**
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by teto** [comment]
I was surprised the zsh tests worked locally but apparently they just grep for stuff so having fewer newlines doesn't impact them @uvNikita

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -547,10 +548,12 @@ in
 
         # Aliases
         ${aliasesStr}
-
-        # Global Aliases
-        ${globalAliasesStr}
```
Comment (sumnerevans) [inline modules/programs/zsh.nix:552]: Can we remove the definition of `globalAliasesStr` from above in the file?

File: modules/programs/zsh.nix [hunk 2]
```diff
@@ -460,44 +460,45 @@ in
...
         # as all $fpath entries will be traversed again.
         ${optionalString (cfg.enableCompletion && !cfg.oh-my-zsh.enable && !cfg.prezto.enable)
           cfg.completionInit
         }
-
-        ${optionalString cfg.enableAutosuggestions
-          "source ${pkgs.zsh-autosuggestions}/share/zsh-autosuggestions/zsh-autosuggestions.zsh"
+        ${optionalString cfg.enableAutosuggestions "
+          source ${pkgs.zsh-autosuggestions}/share/zsh-autosuggestions/zsh-autosuggestions.zsh
+		"
```
Comment (sumnerevans) [inline modules/programs/zsh.nix:501]: This formatting looks a little suspect...

File: modules/programs/zsh.nix (hunk 1 above)
Comment (teto) [reply modules/programs/zsh.nix:552]: good catch

File: modules/programs/zsh.nix (hunk 2 above)
Comment (teto) [reply modules/programs/zsh.nix]: I've restored the version from master

Comment (teto) [reply modules/programs/zsh.nix]: @sumnerevans is that betteR ?

**Review by otavio — APPROVED** [review]

**Review by sumnerevans — APPROVED** [review]

**Pull request merged**